pyteal/compiler/compiler.py

```diff
@@ -46,10 +46,22 @@ def __init__(
         mode: Mode = Mode.Signature,
         version: int = DEFAULT_PROGRAM_VERSION,
         optimize: OptimizeOptions = None,
+        frame_pointers: Optional[bool] = None,
     ) -> None:
         self.mode = mode
         self.version = version
         self.optimize = optimize if optimize is not None else OptimizeOptions()
+        self.use_frame_pointer: bool
+
+        if frame_pointers is None:
+            self.use_frame_pointer = self.version >= FRAME_POINTER_VERSION
+        else:
+            if frame_pointers and self.version < FRAME_POINTER_VERSION:
+                raise TealInputError(
+                    f"Try to use frame pointer with an insufficient version {self.version}."
```

Comment:
I don't think there's a test that triggers this error. Could you add one?

Reply:
yes, add one test in compiler_test